src/rules/no_empty_character_class.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020 the Deno authors. All rights reserved. MIT license.
+use super::Context;
+use super::LintRule;
+use swc_ecma_ast::Regex;
+use swc_ecma_visit::Node;
+use swc_ecma_visit::Visit;
+
+pub struct NoEmptyCharacterClass;
+
+impl LintRule for NoEmptyCharacterClass {
+  fn new() -> Box<Self> {
+    Box::new(NoEmptyCharacterClass)
+  }
+
+  fn code(&self) -> &'static str {
+    "noEmptyCharacterClass"
+  }
+
+  fn lint_module(&self, context: Context, module: swc_ecma_ast::Module) {
+    let mut visitor = NoEmptyCharacterClassVisitor::new(context);
+    visitor.visit_module(&module, &module);
+  }
+}
+
+pub struct NoEmptyCharacterClassVisitor {
+  context: Context,
+}
+
+impl NoEmptyCharacterClassVisitor {
+  pub fn new(context: Context) -> Self {
+    Self { context }
+  }
+}
+
+impl Visit for NoEmptyCharacterClassVisitor {
+  fn visit_regex(&mut self, regex: &Regex, _parent: &dyn Node) {
+    let regex_literal = format!("/{}/", regex.exp);
+    let rule_regex =
+      regex::Regex::new(r"^/([^\[]|\.|\[([^\\\]]|\\.)+\])*/[gimuys]*$")
```

Comment:
make this regex a `lazy_static` and please add comment describing this regex as well as linking to eslint rule

Reply:
thanks for the notice , done!